samples/cellular/nrf_cloud_multi_service/src/provisioning_support.c

```diff
@@ -0,0 +1,184 @@
+		LOG_INF("Provisioning library requests normal mode");
+
+		/* Connect all interfaces since we are done provisioning. */
+		conn_mgr_all_if_connect(true);
+
+		/* Wait for network readiness to be re-established before returning. */
+		LOG_DBG("Waiting for network up");
+		k_event_wait(&prov_events, NETWORK_UP, false, K_FOREVER);
+
+		LOG_DBG("Network is up.");
+	} else if (new_mode == LTE_LC_FUNC_MODE_OFFLINE ||
+		   new_mode == LTE_LC_FUNC_MODE_DEACTIVATE_LTE) {
+		LOG_INF("Provisioning library requests offline mode");
+		/* Disconnect all interfaces to allow cert installation. */
+		conn_mgr_all_if_disconnect(true);
+
+		/* Wait for network disconnection before returning. */
+		LOG_DBG("Waiting for network down.");
+		k_event_wait(&prov_events, NETWORK_DOWN, false, K_FOREVER);
+
+		LOG_DBG("Network is down.");
+	}
+
+	return fmode;
+}
+
+/* Delayable work item for marking provisioning as idle. */
+static void mark_provisioning_idle_work_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	LOG_INF("Provisioning is idle.");
+	k_event_post(&prov_events, PROVISIONING_IDLE);
+}
+static K_WORK_DELAYABLE_DEFINE(provisioning_idle_work, mark_provisioning_idle_work_fn);
+
+/* Called by the provisioning library when each provisioning attempt starts, stops, or finishes
+ * with reboot required.
+ */
+static void device_mode_cb(enum nrf_provisioning_event event, void *user_data)
+{
+	ARG_UNUSED(user_data);
+
+	switch (event) {
+	case NRF_PROVISIONING_EVENT_START:
+		/* Called when the provisioning library begins checking for provisioning commands.
+		 */
+		LOG_DBG("NRF_PROVISIONING_EVENT_START");
+
+		/* Mark provisioning as active. */
+		LOG_INF("Provisioning is active.");
+		k_event_clear(&prov_events, PROVISIONING_IDLE);
+		break;
+	case NRF_PROVISIONING_EVENT_STOP:
+		/* Called when the current provisioning command check has completed. */
+		LOG_DBG("NRF_PROVISIONING_EVENT_STOP");
+
+		/* Mark provisioning as idle after a small delay.
+		 * This delay is to prevent knocking if the provisioning library performs an
+		 * immediate retry.
+		 */
+		k_work_reschedule(&provisioning_idle_work, K_SECONDS(5));
+		break;
+	case NRF_PROVISIONING_EVENT_DONE:
+		/* Called when there are no further provisioning commands,
+		 * and a reboot is needed.
+		 */
+		LOG_DBG("NRF_PROVISIONING_EVENT_DONE");
+
+		LOG_INF("Provisioning completed.");
+		sample_reboot_normal();
+		break;
+	default:
+		LOG_ERR("Unknown event");
+		break;
+	}
+}
+
+static struct nrf_provisioning_mm_change mmode = { .cb = modem_mode_cb };
+static struct nrf_provisioning_dm_change dmode = { .cb = device_mode_cb };
+
+/* Initialize the provisioning library and start checking for provisioning commands.
+ * Called automatically the first time network connectivity is established.
+ */
+static void start_provisioning(void)
+{
+	LOG_INF("Initializing the nRF Provisioning library...");
+
+	/*
+	 * Pass NULL for dmode callback, since the default one is perfectly fine.
```

Comment:
This comment seems to be no longer corrrect.

Reply:
You are correct, good catch